From 6aa1b835d95482287851e02abd3a406cbd0ef8c7 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 9 Dec 2021 15:14:42 -0800
Subject: [PATCH] timezone: Make shell interpreter overridable in tzselect.ksh

define new macro called KSHELL which can be used to define default shell
use Bash by default

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makeconfig        | 9 +++++++++
 timezone/Makefile | 1 +
 2 files changed, 10 insertions(+)

diff --git a/Makeconfig b/Makeconfig
index 85e00cef94..643724108a 100644
--- a/Makeconfig
+++ b/Makeconfig
@@ -309,6 +309,15 @@ ifndef sysincludedir
 sysincludedir = /usr/include
 endif
 
+# The full path name of a Posix-compliant shell, preferably one that supports
+# the Korn shell's 'select' statement as an extension.
+# These days, Bash is the most popular.
+# It should be OK to set this to /bin/sh, on platforms where /bin/sh
+# lacks 'select' or doesn't completely conform to Posix, but /bin/bash
+# is typically nicer if it works.
+ifndef KSHELL
+KSHELL = /bin/bash
+endif
 
 # Commands to install files.
 ifndef INSTALL_DATA
diff --git a/timezone/Makefile b/timezone/Makefile
index d7acb387ba..ec79326f66 100644
--- a/timezone/Makefile
+++ b/timezone/Makefile
@@ -136,6 +136,7 @@ $(objpfx)tzselect: tzselect.ksh $(common-objpfx)config.make
 	    -e '/TZVERSION=/s|see_Makefile|"$(version)"|' \
 	    -e '/PKGVERSION=/s|=.*|="$(PKGVERSION)"|' \
 	    -e '/REPORT_BUGS_TO=/s|=.*|="$(REPORT_BUGS_TO)"|' \
+	    -e 's|#!/bin/bash|#!$(KSHELL)|g' \
 	    < $< > $@.new
 	chmod 555 $@.new
 	mv -f $@.new $@
